clients/src/main/java/org/apache/kafka/common/requests/AbstractResponse.java

```diff
@@ -57,6 +59,10 @@ public ByteBuffer serialize(ApiKeys apiKey, short version, int correlationId) {
         return Collections.singletonMap(error, 1);
     }
 
+    protected Map<Errors, Integer> errorCounts(Stream<Errors> errors) {
+        return errors.collect(Collectors.groupingBy(e -> e, Collectors.summingInt(e -> 1)));
```

Comment:
@tombentley Have we done any testing to verify that this way of doing things is no slower than the old way?